app/models/manageiq/providers/redhat/infra_manager/provision/state_machine.rb

```diff
@@ -69,6 +69,17 @@ def poll_destination_powered_off_in_provider
     end
   end
 
+  def autostart_destination
+    if get_option(:vm_auto_start)
+      message = "Starting"
+      _log.info("#{message} #{for_destination}")
+      update_and_notify_parent(:message => message)
+      get_provider_destination.start { |action| action.use_cloud_init(true) if phase_context[:boot_with_cloud_init] }
```

Comment:
@bdunne We need tests to validate that `use_cloud_init` is set (or not) based on the value of `phase_context[:boot_with_cloud_init]`.